### PR TITLE
ATOM-001-Refactoring of domain object and token creation

### PR DESCRIPTION
**Pull request by amarendraksahoo**
1. JWT token implementation.
2. Code Refactoring.

### CODE REVIEW

File: src/main/java/org/ioak/curate/controller/UserController.java [hunk 1]
```diff
@@ -15,7 +15,6 @@
     public void delete (@PathVariable("name") String name) {
         service.delete(name);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/controller/UserController.java]: make delete methods always based on id

File: src/main/java/org/ioak/curate/repository/UserRepository.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import org.springframework.data.mongodb.repository.MongoRepository;
 
 public interface UserRepository extends MongoRepository<User, String> {
-    User findByUsername(String username);
-    void deleteByUsername(String username);
+    User findByUserName(String username);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/repository/UserRepository.java]: username => name

File: src/main/java/org/ioak/curate/repository/UserRepository.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
...
+    void deleteByUserName(String username);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/repository/UserRepository.java]: delete by id

File: src/main/java/org/ioak/curate/service/BookmarkService.java [hunk 4]
```diff
@@ -17,8 +17,8 @@
         return repository.findAll();
     }
 
-    public Bookmark getBookmark(String username) {
-        return repository.findByUsername(username).orElse(null);
+    public Bookmark getBookmark(String userId) {
+        return repository.findByUserId(userId).orElse(null);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/service/BookmarkService.java]: 1. this should return a list of bookmarks
2. getBookmark => getBookmarkByUser

File: src/main/java/org/ioak/curate/service/NoteService.java [hunk 5]
```diff
@@ -30,7 +30,7 @@ public void delete(Note note) {
     }
 
     public void deleteByNoteid(String noteid) {
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/service/NoteService.java]: deleteByNoteid => delete

File: src/main/java/org/ioak/curate/service/UserService.java [hunk 6]
```diff
@@ -3,50 +3,28 @@
...
-        return hashMap;
-    }*/
-
     public User save(User user) {
         return repository.save(user);
     }
 
     public User getByUsername(String name) {
-        return repository.findByUsername(name);
+        return repository.findByUserName(name);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/service/UserService.java]: getByUsername => getByName

File: src/main/java/org/ioak/curate/service/UserService.java [hunk 7]
```diff
@@ -3,50 +3,28 @@
...
     }
 
     public User getByUserId(String userId) {
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/service/UserService.java]: getByUserId => getById

File: src/main/java/org/ioak/curate/service/UserService.java [hunk 8]
```diff
@@ -3,50 +3,28 @@
...
         return repository.findById(userId).orElse(null);
     }
 
     public void delete(String name) {
-        repository.deleteByUsername(name);
+        repository.deleteByUserName(name);
```
Comment (ioak-dev) [inline src/main/java/org/ioak/curate/service/UserService.java]: public void delete(String name) => public void delete(String id) {

 repository.deleteByUserName(name) =>  repository.deleteById(id)

File: src/main/java/org/ioak/curate/controller/UserController.java (hunk 1 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/controller/UserController.java]: Updated.

File: src/main/java/org/ioak/curate/repository/UserRepository.java (hunk 3 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/repository/UserRepository.java]: Updated.

File: src/main/java/org/ioak/curate/service/BookmarkService.java (hunk 4 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/service/BookmarkService.java]: Updated.

File: src/main/java/org/ioak/curate/service/NoteService.java (hunk 5 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/service/NoteService.java]: Updated.

File: src/main/java/org/ioak/curate/service/UserService.java (hunk 6 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/service/UserService.java]: Updated.

File: src/main/java/org/ioak/curate/service/UserService.java (hunk 7 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/service/UserService.java]: Updated.

File: src/main/java/org/ioak/curate/service/UserService.java (hunk 8 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/service/UserService.java]: Updated.

File: src/main/java/org/ioak/curate/repository/UserRepository.java (hunk 2 above)
Comment (amarendraksahoo) [reply src/main/java/org/ioak/curate/repository/UserRepository.java]: Updated.

**Pull request merged**